File: VB/LayoutViewCustomButtonContent/LayoutViewCustomButtonInfoRegistrator.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraGrid.Registrator
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Views.Base.ViewInfo
Imports DevExpress.XtraGrid.Views.Base.Handler
Imports DevExpress.LookAndFeel

Namespace CustomButtonsLayoutView
	Public Class LayoutViewCustomButtonInfoRegistrator
		Inherits LayoutViewInfoRegistrator
		Public Overrides ReadOnly Property ViewName() As String
			Get
				Return "LayoutViewCustomButton"
			End Get
		End Property
		Public Overrides Function CreateView(ByVal grid As GridControl) As BaseView
			Return New LayoutViewCustomButton(TryCast(grid, GridControl))
		End Function
		Public Overrides Function CreateViewInfo(ByVal view As BaseView) As BaseViewInfo
			Return New LayoutViewCustomButtonViewInfo(TryCast(view, LayoutViewCustomButton))
		End Function
		Public Overrides Function CreateHandler(ByVal view As BaseView) As BaseViewHandler
			Return New LayoutViewCustomButtonHandler(TryCast(view, LayoutViewCustomButton))
		End Function
		Public Overrides Function PaintStyleByLookAndFeel(ByVal lookAndFeel As UserLookAndFeel, ByVal name As String) As ViewPaintStyle
			Return New LayoutViewCustomButtonSkinPaintStyle()
		End Function
	End Class
End Namespace

File: CS/LayoutViewCustomButtonContent/LayoutViewCustomButtonInfoRegistrator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraGrid.Registrator;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.Base.ViewInfo;
using DevExpress.XtraGrid.Views.Base.Handler;
using DevExpress.LookAndFeel;

namespace CustomButtonsLayoutView
{
    public class LayoutViewCustomButtonInfoRegistrator : LayoutViewInfoRegistrator
    {
        public override string ViewName { get { return "LayoutViewCustomButton"; } }
        public override BaseView CreateView(GridControl grid) { return new LayoutViewCustomButton(grid as GridControl); }
        public override BaseViewInfo CreateViewInfo(BaseView view) { return new LayoutViewCustomButtonViewInfo(view as LayoutViewCustomButton); }
        public override BaseViewHandler CreateHandler(BaseView view) { return new LayoutViewCustomButtonHandler(view as LayoutViewCustomButton); }
        public override ViewPaintStyle PaintStyleByLookAndFeel(UserLookAndFeel lookAndFeel, string name) { return new LayoutViewCustomButtonSkinPaintStyle(); }
    }
}
